Detections/AuditLogs/NRT_AuthenticationMethodsChangedforVIPUsers.yaml

```diff
@@ -0,0 +1,37 @@
+id: 29e99017-e28d-47be-8b9a-c8c711f8a903
+name: NRT Authentication Methods Changed for VIP Users
+description: |
+  'Identifies authentication methods being changed for a list of VIP users watchlist. This could be an indication of an attacker adding an auth method to the account so they can have continued access.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+tactics:
+  - Persistence
+relevantTechniques:
+  - T1098
+tags:
+  - AADSecOpsGuide
+query: |
+  let security_info_actions = dynamic(["User registered security info", "User changed default security info", "User deleted security info", "Admin updated security info", "User reviewed security info", "Admin deleted security info", "Admin registered security info"]);
+  let VIPUsers = _GetWatchlist('VIPUsers') | distinct ["User Principal Name"];
+  AuditLogs
+  | where Category =~ "UserManagement"
+  | where ActivityDisplayName in (security_info_actions)
+  | extend Initiator = tostring(InitiatedBy.user.userPrincipalName)
+  | extend IP = tostring(InitiatedBy.user.ipAddress)
+  | extend Target = tolower(tostring(TargetResources[0].userPrincipalName))
+  | where Target in (VIPUsers)
+  | summarize Start=min(TimeGenerated), End=max(TimeGenerated), Actions = make_set(ResultReason, MaxSize=8), Targets=make_set(Target, MaxSize=256) by Initiator, IP, Result
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: FullName
+        columnName: Target
```

Comment:
The entity mapping does not support multiple values that are part of an array. Seems like the Target column might have multiple values due to make_set. Can you please have a look.

Reply:
@aprakash13 - Have expanded the query for 1-1 mapping.
However, the error looks like more of a column not being recognized which is present in watchlist.
Like to know your views. 

https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=41220&view=logs&j=2bbe2297-1b5b-54f3-6060-67e7b289f6c3&t=293f4a23-e36d-5efa-3a4e-ea46d4ab6d83&l=27